### ISSUE TITLE
Unable to push images to registry exposed via unsecure route

### ISSUE THREAD

**Issue by mfojtik** [openshift/origin#12260]
Following the steps described in:

https://docs.openshift.org/latest/install_config/registry/securing_and_exposing_registry.html#access-insecure-registry-by-exposing-route

```
$ oc project default
$ oc expose service docker-registry --hostname=registry.local --as=system:admin
$ docker login -u test-admin -p $(oc whoami -t) -e foo@com registry.local:80
$ oc project test
$ docker tag docker.io/openshift/origin-pod registry.local:80/test/pod:latest
The push refers to a repository [registry.local:80/test/pod]
f31aad019eea: Pushing [==================================================>] 1.172 MB
unauthorized: authentication required
```

Inspecting the registry logs:

```
time="2016-12-14T14:27:24.276470237Z" level=debug msg="filesystem.Writer(\"/docker/registry/v2/repositories/test/pod/_uploads/d58b9681-0ec6-4169-8f63-51dd3ea30bf0/data\", false)" go.version=go1.7.4 http.request.host=registry.local http.request.id=f76a5e08-0d90-4354-973d-88b0d44d5a27 http.request.method=POST http.request.remoteaddr=192.168.64.3 http.request.uri="/v2/test/pod/blobs/uploads/" http.request.useragent="docker/1.10.3 go/go1.6.3 git-commit/cb079f6-unsupported kernel/3.10.0-327.el7.x86_64 os/linux arch/amd64" instance.id=fa344146-8139-45cd-ad75-c221f6fd9189 trace.duration=41.42µs trace.file="/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/openshift/origin/vendor/github.com/docker/distribution/registry/storage/driver/base.(*Base).Writer" trace.id=99d3c278-b516-4474-aaf8-f3b463bc7afb trace.line=124 vars.name="test/pod"
time="2016-12-14T14:27:24.27702426Z" level=debug msg="filesystem.PutContent(\"/docker/registry/v2/repositories/test/pod/_uploads/d58b9681-0ec6-4169-8f63-51dd3ea30bf0/hashstates/sha256/0\")" go.version=go1.7.4 http.request.host=registry.local http.request.id=f76a5e08-0d90-4354-973d-88b0d44d5a27 http.request.method=POST http.request.remoteaddr=192.168.64.3 http.request.uri="/v2/test/pod/blobs/uploads/" http.request.useragent="docker/1.10.3 go/go1.6.3 git-commit/cb079f6-unsupported kernel/3.10.0-327.el7.x86_64 os/linux arch/amd64" instance.id=fa344146-8139-45cd-ad75-c221f6fd9189 trace.duration=459.817µs trace.file="/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/openshift/origin/vendor/github.com/docker/distribution/registry/storage/driver/base.(*Base).PutContent" trace.id=7235d30f-1a9e-441c-b37b-6b26067a8f9a trace.line=95 vars.name="test/pod"
time="2016-12-14T14:27:24.277182332Z" level=info msg="response completed" go.version=go1.7.4 http.request.host=registry.local http.request.id=f76a5e08-0d90-4354-973d-88b0d44d5a27 http.request.method=POST http.request.remoteaddr=192.168.64.3 http.request.uri="/v2/test/pod/blobs/uploads/" http.request.useragent="docker/1.10.3 go/go1.6.3 git-commit/cb079f6-unsupported kernel/3.10.0-327.el7.x86_64 os/linux arch/amd64" http.response.duration=7.599583ms http.response.status=202 http.response.written=0 instance.id=fa344146-8139-45cd-ad75-c221f6fd9189
172.17.0.1 - - [14/Dec/2016:14:27:24 +0000] "POST /v2/test/pod/blobs/uploads/ HTTP/1.1" 202 0 "" "docker/1.10.3 go/go1.6.3 git-commit/cb079f6-unsupported kernel/3.10.0-327.el7.x86_64 os/linux arch/amd64"
172.17.0.1 - - [14/Dec/2016:14:27:24 +0000] "GET /healthz HTTP/1.1" 200 0 "" "Go-http-client/1.1"
172.17.0.1 - - [14/Dec/2016:14:27:24 +0000] "GET /healthz HTTP/1.1" 200 0 "" "Go-http-client/1.1"
time="2016-12-14T14:27:24.322551897Z" level=debug msg="authorizing request" go.version=go1.7.4 http.request.host=registry.local http.request.id=a8c634ba-c75f-4e81-b43c-6249876ca272 http.request.method=PATCH http.request.remoteaddr=192.168.64.3 http.request.uri="/v2/test/pod/blobs/uploads/d58b9681-0ec6-4169-8f63-51dd3ea30bf0?_state=cLfDYLlm9CciAQfkt1ne8mWb-lQPl76Su-3Jz2mZsex7Ik5hbWUiOiJ0ZXN0L3BvZCIsIlVVSUQiOiJkNThiOTY4MS0wZWM2LTQxNjktOGY2My01MWRkM2VhMzBiZjAiLCJPZmZzZXQiOjAsIlN0YXJ0ZWRBdCI6IjIwMTYtMTItMTRUMTQ6Mjc6MjQuMjc1ODA5MzI0WiJ9" http.request.useragent="docker/1.10.3 go/go1.6.3 git-commit/cb079f6-unsupported kernel/3.10.0-327.el7.x86_64 os/linux arch/amd64" instance.id=fa344146-8139-45cd-ad75-c221f6fd9189 vars.name="test/pod" vars.uuid=d58b9681-0ec6-4169-8f63-51dd3ea30bf0
time="2016-12-14T14:27:24.322702188Z" level=error msg="error authorizing context: authorization header required" go.version=go1.7.4 http.request.host=registry.local http.request.id=a8c634ba-c75f-4e81-b43c-6249876ca272 http.request.method=PATCH http.request.remoteaddr=192.168.64.3 http.request.uri="/v2/test/pod/blobs/uploads/d58b9681-0ec6-4169-8f63-51dd3ea30bf0?_state=cLfDYLlm9CciAQfkt1ne8mWb-lQPl76Su-3Jz2mZsex7Ik5hbWUiOiJ0ZXN0L3BvZCIsIlVVSUQiOiJkNThiOTY4MS0wZWM2LTQxNjktOGY2My01MWRkM2VhMzBiZjAiLCJPZmZzZXQiOjAsIlN0YXJ0ZWRBdCI6IjIwMTYtMTItMTRUMTQ6Mjc6MjQuMjc1ODA5MzI0WiJ9" http.request.useragent="docker/1.10.3 go/go1.6.3 git-commit/cb079f6-unsupported kernel/3.10.0-327.el7.x86_64 os/linux arch/amd64" instance.id=fa344146-8139-45cd-ad75-c221f6fd9189 vars.name="test/pod" vars.uuid=d58b9681-0ec6-4169-8f63-51dd3ea30bf0
172.17.0.1 - - [14/Dec/2016:14:27:24 +0000] "PATCH /v2/test/pod/blobs/uploads/d58b9681-0ec6-4169-8f63-51dd3ea30bf0?_state=cLfDYLlm9CciAQfkt1ne8mWb-lQPl76Su-3Jz2mZsex7Ik5hbWUiOiJ0ZXN0L3BvZCIsIlVVSUQiOiJkNThiOTY4MS0wZWM2LTQxNjktOGY2My01MWRkM2VhMzBiZjAiLCJPZmZzZXQiOjAsIlN0YXJ0ZWRBdCI6IjIwMTYtMTItMTRUMTQ6Mjc6MjQuMjc1ODA5MzI0WiJ9 HTTP/1.1" 401 196 "" "docker/1.10.3 go/go1.6.3 git-commit/cb079f6-unsupported kernel/3.10.0-327.el7.x86_64 os/linux arch/amd64"
```

**Comment by mfojtik** [comment]
@miminar PTAL (it might be I'm doing something wrong)

**Comment by miminar** [comment]
This is a duplicate of  #11820

**Comment by miminar** [comment]
Another fix for this awaiting review is #11391
